### PR TITLE
NKAI improve defense and some fixes

### PR DESCRIPTION
**Pull request by nullkiller**
AI should now more eagerly destroy heroes causing treat on AI towns.
AI considers FORT level when analyzing enemy towns value.
AI still can consider that enemy strength is too high and just give up.

### CODE REVIEW

File: AI/Nullkiller/Engine/Nullkiller.cpp [hunk 1]
```diff
@@ -221,6 +222,7 @@ void Nullkiller::makeTurn()
 	boost::lock_guard<boost::mutex> sharedStorageLock(AISharedStorage::locker);
 
 	const int MAX_DEPTH = 10;
+	const int FAST_TASK_MINIMAL_PRIORITY = 0.7;
```
Comment (DjWarmonger) [inline AI/Nullkiller/Engine/Nullkiller.cpp:225]: Build failed because of that line, surely not what you intended.

Comment (nullkiller) [reply AI/Nullkiller/Engine/Nullkiller.cpp]: Yes, for sure.

Comment (rilian-la-te) [reply AI/Nullkiller/Engine/Nullkiller.cpp]: @nullkiller maybe use constexpr int?

File: AI/Nullkiller/Engine/PriorityEvaluator.cpp [hunk 2]
```diff
@@ -238,7 +252,17 @@ uint64_t RewardEvaluator::getArmyReward(
...
-		return target->tempOwner == PlayerColor::NEUTRAL ? 1000 : 10000;
+	{
+		auto town = dynamic_cast<const CGTownInstance *>(target);
+		auto fortLevel = town->fortLevel();
+		auto booster = isAnotherAi(town, *ai->cb) ? 1 : 2;
+
+		if(fortLevel < CGTownInstance::CITADEL)
+			return town->hasFort() ? booster * 500 : 0;
+		else
+			return fortLevel == booster * (CGTownInstance::CASTLE ? 5000 : 2000);
```
[the remark this answers is not in the document]
Comment (DjWarmonger) [reply AI/Nullkiller/Engine/PriorityEvaluator.cpp:263]: Also, name `booster` is somewhat unclear.

[the remark this answers is not in the document]
Comment (nullkiller) [reply AI/Nullkiller/Engine/PriorityEvaluator.cpp:263]: Ouch

**Pull request merged**